app/models/ems_refresh/save_inventory_infra.rb

```diff
@@ -159,6 +159,10 @@ def save_hosts_inventory(ems, hashes, target = nil)
           #   and the new hostname is not an ip address
           h[:name] = h[:hostname] if h[:name] =~ ip_part && !(h[:hostname] =~ ip_whole)
 
+          # Update the name to be found.name if it has the same ems_ref and the name
+          # already has a '- int' suffix to work around duplicate hostnames
+          h[:name] = found.name if found.name =~ /#{h[:name]} - \d+$/
```

Comment:
Yeah what I was worried about is comparing `h[:name]` and `found.name` first when originally we might have set `h[:name] = h[:hostname]`, but I might be completely missing the point of that second block of code.  If we have a name already can we ignore `h[:name] = h[:hostname] if h[:name] =~ ip_part && !(h[:hostname] =~ ip_whole)` entirely?